config/initializers/npm.rb

```diff
@@ -0,0 +1,2 @@
+# prevent rails/bundler from starting itself until npm is installed
+system 'npm install' if Rails.env.development? || Rails.env.test?
```

Comment:
Rails ships with yarn so I think it would be best if you use that over npm. It works more or less identically to npm (plus it's usually faster to run) and uses the same `pacakge.json` so there's little reason to add npm as an additional dependency.

Reply:
Also, running npm install every time we run the app or tests is going to slow things down quite significantly. Could this be moved outside the application to somewhere else? Yarn has a command `yarn check --integrity` which might be faster.

Reply:
It might be helpful to check how Rails currently uses `bunder` to check for ruby dependencies at runtime.